tests/warn/i22232.scala

```diff
@@ -0,0 +1,19 @@
+class C
+object C:
+  extension (c: C) def equals(that: Any): Boolean = false // warn
+
+object X:
+  class C
+  opaque type D <: C = C
+  object D:
+    extension (d: D) def equals(that: Any): Boolean = false // warn
+
+object Upperbound:
+  opaque type IArray[+T] <: String = String
+  extension (arr: IArray[Byte]) def length: Int = 0 // warn
+  extension [T <: Int](arr: T) def length: Int = 0 // nowarn
```

Comment:
Won't harm to add this extra test
```suggestion
  extension [T <: IArray[Byte]](arr: T) def length: Int = 0 // warn
```